pkg/cli/cli.go

```diff
@@ -0,0 +1,163 @@
+	"github.com/cyberark/conjur-oss-suite-release/pkg/changelog"
+	"github.com/cyberark/conjur-oss-suite-release/pkg/github"
+	"github.com/cyberark/conjur-oss-suite-release/pkg/http"
+	"github.com/cyberark/conjur-oss-suite-release/pkg/repositories"
+	"github.com/cyberark/conjur-oss-suite-release/pkg/template"
+)
+
+// Options represents the command line values a user can pass in
+type Options struct {
+	APIToken           string
+	Date               time.Time
+	OutputFilename     string
+	OutputType         string
+	RepositoryFilename string
+	Version            string
+}
+
+type templateInfo struct {
+	TemplateName   string
+	OutputFileName string
+}
+
+const defaultOutputType = "changelog"
+const defaultRepositoryFilename = "repositories.yml"
+const defaultVersionString = "Unreleased"
+
+
+// RunParser kicks off the process for writing a new changelog
+// 1. Set the output filename if one is not set
+// 2. Collect data on each component as specified
+// 3. Build a unified changelog with each component
+// 4. Write a new changelog based on the appropriate template
+func RunParser(options Options) error {
+	templateInfo, err := getTemplateInfo(options.OutputType)
+	if err != nil {
+		return err
+	}
+	options.setOutputFilename(templateInfo.OutputFileName)
+
+	log.Printf("Parsing linked repositories...")
+	repoConfig, err := repositories.NewConfig(options.RepositoryFilename)
+	if err != nil {
+		return err
+	}
+
+	if options.OutputType == "unreleased" {
+		// This is an in-place operation
+		repoConfig.SelectUnreleased()
+	}
+
+	log.Printf("Collecting changelogs...")
+	httpClient := http.NewClient()
+
+	githubAPIToken := options.APIToken
+	if githubAPIToken == "" {
+		githubAPIToken = os.Getenv("GITHUB_TOKEN")
+	}
+	httpClient.AuthToken = githubAPIToken
+
+	components, err := github.CollectComponents(repoConfig, httpClient)
+	if err != nil {
+		return fmt.Errorf("ERROR: %v", err)
+	}
+
+	// Combine all changelogs into a single array to generate the unified changelog
+	changelogs := []*changelog.VersionChangelog{}
+	for _, component := range components {
+		changelogs = append(changelogs, component.Changelogs...)
+	}
+	unifiedChangelog := changelog.NewUnifiedChangelog(changelogs...)
+
+	// TODO: Should the date be something defined in yml or the date of tag?
+	if options.Date.IsZero() {
+		options.Date = time.Now()
+	}
+
+	templateData := template.ReleaseSuite{
+		// TODO: Suite version should probably be read from some file
+		Version:          options.Version,
+		Date:             options.Date,
+		Components:       components,
+		UnifiedChangelog: unifiedChangelog.String(),
+	}
+
+	tmpl := template.New("templates")
+	err = tmpl.WriteChangelog(templates[options.OutputType].TemplateName,
+		templateData,
+		options.OutputFilename)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("Changelog parser completed!")
+	return err
+}
+
+// HandleInput parses command line values and stores them within an options struct
+func (options *Options) HandleInput() {
```

Comment:
Why not just use `Parse` for the name like the `flag` package (or `ParseFlags`)?